pkg/s3select/select.go

```diff
@@ -220,25 +231,35 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
 		if condition {
 			// if its an asterix we just print everything in the row
 			if reqColNames[0] == "*" && functionNames[0] == "" {
-				rowStruct := &Row{
-					record: reader.printAsterix(record) + "\n",
+				switch inputType.(type) {
+				case *CSVInput:
+					rowStruct := &Row{
+						record: printAsterix(convertToSlice(columnsMap, individualRecord, string(out)), inputType) + "\n",
```

Comment:
name can be different, something like `printAllColumns` instead of `printAsterix`

Reply:
Changed